apps/session-manager/src/controllers/__tests__/authentication.spec.ts

```diff
@@ -1178,3 +1187,61 @@ describe("AuthenticationController#acs LV Notify user login", () => {
     );
   });
 });
+
+describe("AuthenticationController#acsTest", () => {
+  const acsSpyOn = vi.spyOn(AuthController, "acs");
+  beforeEach(() => {
+    vi.clearAllMocks();
+    resetMock(res as unknown as ReturnType<typeof mockRes>);
+  });
+  afterAll(() => {
+    acsSpyOn.mockRestore();
+  });
+  test("should return ResponseSuccessJson with a valid token if acs succeeded", async () => {
+    const expectedToken = "token-111-222";
+    acsSpyOn.mockReturnValueOnce(async (_: unknown) =>
+      ResponsePermanentRedirect(getClientProfileRedirectionUrl(expectedToken)),
+    );
+    const response = await acsTest(validUserPayload)({
+      ...dependencies,
+      clientProfileRedirectionUrl,
+    })();
+    expect(response).toEqual(
+      E.right(
+        toExpectedResponse(ResponseSuccessJson({ token: expectedToken })),
+      ),
+    );
+  });
+
+  test("should return the same response of acs if is different from SuccessPermanentRedirect", async () => {
+    const expectedResponse = ResponseErrorValidation(
+      "Validation error",
+      "Validation error message",
+    );
+    acsSpyOn.mockReturnValueOnce(async (_: unknown) => expectedResponse);
+    const response = await acsTest(validUserPayload)({
+      ...dependencies,
+      clientProfileRedirectionUrl,
+    })();
+
+    expect(response).toEqual(E.right(toExpectedResponse(expectedResponse)));
+  });
+
+  test("should return ResponseErrorInternal if the token is missing", async () => {
+    acsSpyOn.mockReturnValueOnce(async (_: unknown) =>
+      ResponsePermanentRedirect({
+        href: "https://invalid-url",
+      } as ValidUrl),
+    );
+    const response = await acsTest(validUserPayload)({
+      ...dependencies,
+      clientProfileRedirectionUrl,
+    })();
+
+    expect(response).toEqual(
+      E.right(
+        toExpectedResponse(ResponseErrorInternal("Unexpected redirection url")),
+      ),
+    );
+  });
```

Comment:
I would suggest to also add this test to increase coverage
```suggestion
  });
    test("should return ResponseErrorInternal if the token decode fails", async () => {
    acsSpyOn.mockReturnValueOnce(async (_: unknown) =>
      ResponsePermanentRedirect(getClientProfileRedirectionUrl("")),
    );
    const response = await acsTest(validUserPayload)({
      ...dependencies,
      clientProfileRedirectionUrl,
    })();
    expect(response).toEqual(
      E.right(
        toExpectedResponse({
          detail: expect.stringContaining("Decode Error:"),
          kind: "IResponseErrorInternal",
        }),
      ),
    );
  });
```

Reply:
Done in 4f4ed7c